test/files.spec.js

```diff
@@ -87,6 +95,53 @@ describe('.files (the MFS API part)', function () {
       })
     })
 
+    it('files.add without only-hash', (done) => {
+      const content = String(Math.random() + Date.now())
+      const inputFile = createTestFile(content)
+
+      ipfs.files.add([inputFile], { onlyHash: false }, (err, res) => {
+        expect(err).to.not.exist()
+
+        const hash = res[0].hash
+        let retrievedContent = ''
+
+        ipfs.get(hash, (err, res) => {
+          expect(err).to.not.exist()
+
+          res.pipe(through.obj((file, encoding, next) => {
+            file.content.pipe(concat(retrieved => {
+              retrievedContent += retrieved.toString()
+              next()
+            }))
+          }, () => {
+            expect(content).to.equal(retrievedContent)
+            done()
+          }))
+        })
+      })
+    })
+
+    it.only('files.add with only-hash', (done) => {
+      const inputFile = createTestFile()
+
+      ipfs.files.add([inputFile], {'only-hash': true}, (err, res) => {
+        expect(err).to.not.exist()
+
+        streamToValue(res, (err, collected) => {
+          expect(err).to.not.exist()
+
+          const hash = collected[0].Hash
+
+          ipfs.files.get(hash, (err, res) => {
```

Comment:
Use `ipfs refs local` for this.

https://github.com/ipfs/js-ipfs/issues/780#issuecomment-285059426